helper/src/main/java/io/github/libxposed/helpers/BaseFinder.java

```diff
@@ -0,0 +1,254 @@
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@SuppressWarnings({"unchecked", "unused"})
+abstract class BaseFinder<T extends Member, S> {
+    @NonNull
+    protected Stream<T> stream;
+
+    BaseFinder(@NonNull Stream<T> stream) {
+        this.stream = stream;
+    }
+
+    public final S filter(@NonNull Predicate<T> predicate) {
+        stream = stream.filter(predicate);
+        return (S) this;
+    }
+
+    /**
+     * Filter methods/constructors if they have public modifier.
+     *
+     * @return this
+     */
+    public final S filterPublic() {
+        return filter(ModifierHelper::isPublic);
+    }
+
+    /**
+     * Filter methods/constructors if they not have public modifier.
+     *
+     * @return this
+     */
+    public final S filterNonPublic() {
+        return filter(ModifierHelper::isNotPublic);
+    }
+
+    /**
+     * Filter methods/constructors if they have protected modifier.
+     *
+     * @return this
+     */
+    public final S filterProtected() {
+        return filter(ModifierHelper::isProtected);
+    }
+
+    /**
+     * Filter methods/constructors if they not have protected modifier.
+     *
+     * @return this
+     */
+    public final S filterNonProtected() {
+        return filter(ModifierHelper::isNotProtected);
+    }
+
+    /**
+     * Filter methods/constructors if they have private modifier.
+     *
+     * @return this
+     */
+    public final S filterPrivate() {
+        return filter(ModifierHelper::isPrivate);
+    }
+
+    /**
+     * Filter methods/constructors if they not have private modifier.
+     *
+     * @return this
+     */
+    public final S filterNonPrivate() {
+        return filter(ModifierHelper::isNotPrivate);
+    }
+
+    /**
+     * Filter methods/constructors if they have native modifier.
+     *
+     * @return this
+     */
+    public final S filterNative() {
+        return filter(ModifierHelper::isNative);
+    }
+
+    /**
+     * Filter methods/constructors if they have native modifier.
+     *
+     * @return this
+     */
+    public final S filterNonNative() {
+        return filter(ModifierHelper::isNotNative);
+    }
+
+    /**
+     * Filter methods/constructors if they are package private.
+     *
+     * @return this
+     */
+    public final S filterPackage() {
+        return filter(ModifierHelper::isPackage);
+    }
+
+    /**
+     * Filter methods/constructors if they are not package private.
+     *
+     * @return this
+     */
+    public final S filterNonPackage() {
+        return filter(ModifierHelper::isNotPackage);
+    }
+
+    /**
+     * Filter methods/constructors by their exception types(Full-matches, cannot be null)
+     *
+     * @param e exception types
+     * @return this
+     */
+    public final S filterExceptionType(Class<?>... e) {
```

Comment:
Confusing naming. This method is named with `by` but the next method has no `by` in its name.